teps/draft-20200715-access-control-via-sa.md

```diff
@@ -0,0 +1,249 @@
+(RBAC and admission controllers/plugins/webhooks among others) that allow administrators to restrict the amount 
+Kubernetes API interaction a controller can do on behalf of a given user.
+
+Similarly, in pre-existing CI/CD system (Jenkins for example), there are similar capabilities in place that allow
+administrators to read/write/execute access permissions on global Jobs/Pipelines (rough equivalent of cluster scoped 
+resources like ClusterTasks) and Jobs/Pipelines within Folders (rough equivalent of namespace scoped resources like 
+Tasks) for specific users or groups of users.    
+
+As part of both
+
+- Continuing the perception Tekton as a true, Kubernetes native CI/CD system, that leverages all the inherent 
+capabilities of Kubernetes
+- Reaching at least equal capability, compared to existing options, across the spectrum of concerns of a CI/CD system
+
+This proposal articulates a path to finer grained authorization/access control around the various Tekton concepts.
+
+### Goals
+
+One goal of this TEP is to define an approach for access control for references from one Tekton object to another in
+a manner that is natural for Kubernetes cluster administrators.
+
+A second goal of this TEP is introduce such access control in a manner that does not immediately disrupt existing 
+deployments of Tekton, given the free access to all approach that currently exists for these Tekton object to Tekton
+object references.
+
+### Non-Goals
+
+Other valid authorization related scenarios have been raised in various community discussions.  This proposal 
+does not attempt to address all of them.
+
+Among these various scenarios, there has also been discussion around introducing OPA (either via the admission webhook 
+for Kubernetes provided by the OPA org, or some other form of integration) into Tekton.  This proposal is not taking
+on how OPA would be integrated / leveraged by Tekton. 
+
+## Requirements
+
+Kubernetes administrators have means to enact control over which ClusterTasks, Tasks, or Pipelines are referenced
+by a PipelineRun or TaskRun.
+
+Kubernetes administrators have a means to opt into such enforcement on a granular basis (per namespace). 
+
+## Proposal
+
+An addition of a Kubernetes Validating Admission Webhook to the current Tekton Webhooks Deployment can 
+serve as the contact point for intercepting any create/update requests for TaskRun and PipelineRun and 
+then see if the ServiceAccount specified (or "default" if non specified) has permission to get/read any
+referenced ClusterTask/Task/Pipeline objects.
+
+A Kubernetes RBAC Subject Access Review (SAR) check will be executed to see if the ServiceAccount in question
+is part of a Role/RoleBinding or ClusterRole/ClusterRoleBinding that allows access to the referenced 
+ClusterTask, Task, or PipelineTask.  If access is denied, the admission webhook will prevent creation/update
```

Comment:
s/PiplineTask/Pipeline? (a PipelineTask isnt a stand alone thing that can be referenced)

Reply:
yep that is what I meant ... will update - thanks!